### PR TITLE
Assign debugCreator for more layers

### PR DESCRIPTION
**Pull request by wangying3426**
A little patch to assign values to layer's debugCreator properties, such as OpacityLayer、ImageFilterLayer etc.

Take ImageFilterLayer as an example.
Now:
```
flutter:  │     │ │
flutter:  │     │ └─child 1: ImageFilterLayer#c2ce6
flutter:  │     │   │ engine layer: ImageFilterEngineLayer#b0b19
flutter:  │     │   │ handles: 2
flutter:  │     │   │ imageFilter: ImageFilter.matrix([-0.7828034697571389,
flutter:  │     │   │   0.6222690155681738, 0.0, 0.0, -0.6222690155681738,
flutter:  │     │   │   -0.7828034697571389, -0.0, -0.0, 0.0, 0.0, 1.0, 0.0,
flutter:  │     │   │   360.7608727987969, 174.08016812834478, 0.0, 1.0],
flutter:  │     │   │   FilterQuality.low)
flutter:  │     │   │
```
After merged: 
```
flutter:  │     │ │
flutter:  │     │ └─child 1: ImageFilterLayer#c2ce6
flutter:  │     │   │ creator: ImageFiltered ← AnimatedBuilder ← RepaintBoundary ←
flutter:  │     │   │   Center ← _BodyBuilder ← MediaQuery ←
flutter:  │     │   │   LayoutId-[<_ScaffoldSlot.body>] ← CustomMultiChildLayout ←
flutter:  │     │   │   AnimatedBuilder ← DefaultTextStyle ← AnimatedDefaultTextStyle ←
flutter:  │     │   │   _InkFeatures-[GlobalKey#b731c ink renderer] ← ⋯
flutter:  │     │   │ engine layer: ImageFilterEngineLayer#b0b19
flutter:  │     │   │ handles: 2
flutter:  │     │   │ imageFilter: ImageFilter.matrix([-0.7828034697571389,
flutter:  │     │   │   0.6222690155681738, 0.0, 0.0, -0.6222690155681738,
flutter:  │     │   │   -0.7828034697571389, -0.0, -0.0, 0.0, 0.0, 1.0, 0.0,
flutter:  │     │   │   360.7608727987969, 174.08016812834478, 0.0, 1.0],
flutter:  │     │   │   FilterQuality.low)
flutter:  │     │   │
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by goderbauer — COMMENTED** [review]
Can you please add a test so we don't accidentally remove this again? Thanks!

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -5196,7 +5216,6 @@ class RenderLeaderLayer extends RenderProxyBox {
         ..offset = offset;
     }
     context.pushLayer(layer!, super.paint, Offset.zero);
-    assert(layer != null);
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/src/rendering/proxy_box.dart:5199]: @wangying3426 This question is still unanswered.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**